gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java

```diff
@@ -588,94 +588,86 @@ public void visitErrorNode(ErrorNode node) {
     }
 
     @Override
-    public void exitInvokeMediatorCall(WUMLParser.InvokeMediatorCallContext ctx) {
-        /*  Implementation is done to directly call the backend and respond.
-            Only supports when "return invoke(EP, message)"
-            "message m = invoke(EP, message)" not supported
-         */
-
-        //Add the call mediator
-        Mediator callMediator = MediatorProviderRegistry.getInstance().getMediator("call");
-
-        ParameterHolder parameterHolder = new ParameterHolder();
-        parameterHolder.addParameter(new Parameter("endpointKey", ctx.Identifier().get(0).getText()));
-        parameterHolder.addParameter(new Parameter("inputMessageParameter", ctx.Identifier().get(1).getText()));
-        parameterHolder.addParameter(new Parameter("integrationKey", integrationName));
-
-        if (ctx.parent instanceof  WUMLParser.MediatorCallContext) {
-            parameterHolder.addParameter(new Parameter("returnVariableKey",
-                    ((WUMLParser.MediatorCallContext) ((WUMLParser.InvokeMediatorCallContext) ctx).parent).Identifier()
-                            .getText()));
-        }
-
-        callMediator.setParameters(parameterHolder);
-
-        dropMediatorFilterAware(callMediator);
-    }
-
-    @Override
     public void exitResourceName(WUMLParser.ResourceNameContext ctx) {
         /* Creating the resource */
         this.currentResource = new Resource(ctx.Identifier().getText());
     }
 
+    /**
+     * Handle events regarding all the data manipulation mediators (including custom mediators)
+     * @param ctx
+     */
     @Override
-    public void exitLogMediatorCall(WUMLParser.LogMediatorCallContext ctx) {
-        Mediator logMediator = MediatorProviderRegistry.getInstance().getMediator(Constants.LOG_MEDIATOR);
-        ParameterHolder parameterHolder = new ParameterHolder();
-        logMediator.setParameters(parameterHolder);
-        //Currently the log only support for log full and custom string input
-        if (ctx.literal() != null) {
-            parameterHolder.addParameter(new Parameter("Log Message:", ctx.literal().getText()));
-            parameterHolder.addParameter(new Parameter(Constants.LEVEL, Constants.LEVEL_CUSTOM));
-        } else {
-            parameterHolder.addParameter(new Parameter(Constants.LEVEL, Constants.LEVEL_FULL));
+    public void exitMediatorCall(WUMLParser.MediatorCallContext ctx) {
+        String mediatorId =  ctx.Identifier().getText();
+        // call mediator is specified in the language as "invoke". This is a special case.
+        if (Constants.INVOKE_STATEMENT.equals(mediatorId)) {
+            mediatorId = Constants.CALL_MEDIATOR_NAME;
         }
 
-        logMediator.setParameters(parameterHolder);
-        dropMediatorFilterAware(logMediator);
+        Mediator mediator = MediatorProviderRegistry.getInstance().getMediator(mediatorId);
+
+        if (mediator != null) {
```

Comment:
Mediators are loaded in runtime, If a name of a custom mediator is given in the NEL config but not the mediator is not registered in the MediatorProviderRegistry, this will return null. 
I think adding a WARN log is a good suggestion, I will do that.
Thanks,